common/setups/rasr/gmm_system.py

```diff
@@ -826,6 +830,46 @@ def sat_recognition(
                     **kwargs,
                 )
 
+    # -------------------- output helpers  --------------------
+
+    def get_gmm_output(
+        self, corpus_key: str, corpus_type: str, step_idx: int, steps: RasrSteps
+    ):
+        """
+        :param corpus_key: corpus name identifier
+        :param corpus_type: corpus used for: train, dev or test
+        :param step: step name in the gmm pipeline: tri, vtln, ...
+        :param iteration: which iteration to take within one step
+        :return GmmOutput:
+        """
+        gmm_output = GmmOutput()
+        gmm_output.crp = self.crp[corpus_key]
+        gmm_output.corpus_file = self.corpora[corpus_key].corpus_file
+        gmm_output.corpus_object = self.corpora[corpus_key]
+        gmm_output.lexicon_file = self.crp[corpus_key].lexicon_config.file
+        gmm_output.lexicon = self.crp[corpus_key].lexicon_config
+        gmm_output.cart_tree = self.crp[
+            corpus_key
+        ].acoustic_model_config.state_tying.file
+        gmm_output.feature_flows = self.feature_flows[corpus_key]
+        gmm_output.features = self.feature_caches[corpus_key]
+        gmm_output.segment_path = self.crp[corpus_key].segment_path
+        gmm_output.allophone_file = self.allophone_files["base"]
+
+        if corpus_type == "dev" or corpus_type == "test":
```

Comment:
Why only set LM for test sets? E.g. for sequence training we also have a LM for the train set.

Reply:
Yes. But this is the GMM output. I thought that for sequence training we normally want a different LM for lattice generation. For example, Librispeech count LM for decoding is 4gram. For sequence training, this is a 2gram. or not?
That is why I did not want to automatically set the LM for training.. but I am open to handle it differently

Reply:
Yes, so for train set a 2gram LM would be exported.

This is not about automatically setting it, but about exporting it if set.